src/imitation/algorithms/preference_comparisons.py

```diff
@@ -735,14 +966,13 @@ def forward(
                 reward model and the target probabilities in `preferences`. Metrics
                 are include accuracy.
         """
-        probs = th.empty(len(fragment_pairs), dtype=th.float32)
-        for i, fragment in enumerate(fragment_pairs):
-            frag1, frag2 = fragment
-            trans1 = rollout.flatten_trajectories([frag1])
-            trans2 = rollout.flatten_trajectories([frag2])
-            rews1 = self._rewards(model, trans1)
-            rews2 = self._rewards(model, trans2)
-            probs[i] = self._probability(rews1, rews2)
+        preference_predictor = PreferencePredictor(
```

Comment:
Make sure preference predictor passes gradients through from the reward network since it's used to calculate the loss.

Reply:
The preference predictor does pass the gradient through the reward network. It just doesn't pass the gradient through the RewardEnsemble since predict_processed_all returns a Numpy array. But this is not a bug since the `EnsembleTrainer` goes through the member networks and calculates the loss for them separately. Hence, for each of the member networks, the gradient will be passed through the loss computation.